corehq/apps/app_manager/models.py

```diff
@@ -4478,18 +4478,14 @@ def check_build_dependencies(self, new_build):
         """
 
         def has_dependencies(build):
-            if isinstance(build, Application):
-                profile = build.profile
-            else:
-                profile = build.get('profile', {})
-
             return bool(
-                profile.get('features', {}).get('dependencies')
+                build.profile.get('features', {}).get('dependencies')
             )
 
         new_build_has_dependencies = has_dependencies(new_build)
 
         last_build = get_latest_build_doc(self.domain, self.id)
+        last_build = ApplicationBase.wrap(last_build) if last_build else None
```

Comment:
nit: Not sure the difference but should this be `cls.wrap`? just like when the copy is created.

Reply:
This is in instance method, so either `self.__class__.wrap` or `ApplicationBase` - I prefer the latter.

Reply:
Will that impact if the function is called by inherited classes instead?

Reply:
Yes, if a subclass calls this method that would mean the ApplicationBase wrap method is still called and not the subclass's `wrap` method. I think this is OK, since we don't use anything except the profile attribute. But that is a good point, maybe it should take the subclass's own implementation of `wrap`?

Reply:
> ApplicationBase wrap method is still called and not the subclass's wrap method

Ya. We would be wrapping data for a different class on ApplicationBase which isn't ideal and would further lead to issues if the subclass adds more properties than ApplicationBase, which will get skipped leading to loss of data in case a save is performed.

Reply:
Good point. I'll update

Reply:
You might check out [wrap_app](https://github.com/dimagi/commcare-hq/blob/873031e815cd886a9d3fca0ec9474094cfb7fc95/corehq/apps/app_manager/dbaccessors.py#L151) for this.

Reply:
@orangejenny What is the exact benefit of using `wrap_app(last_build, self.__class__)` instead of simply `self.__class__.wrap(last_build)`? I see the `wrap_doc` function have the `get_correct_app_class` addition, but that's assuming I don't pass in the class name, which I will do.

Reply:
Probably doesn't matter, then. I had just read the original code when making the suggestion.

Reply:
`wrap_app` looks like a nice catch all for different application and similar docs with a hard failure in case of unexpected scenarios. Definitely good to use when dealing with more complex code.
The class wrap seems safe here since we are passing a build explicitly.